[CI] ReindexDatastreamIndexTransportActionIT testTsdbStartEndSet failing

**Build Scans:**
- [elasticsearch-intake #28971 / part2](https://gradle-enterprise.elastic.co/s/if7txbuea6vfe)
- [elasticsearch-pull-request #94579 / part-2](https://gradle-enterprise.elastic.co/s/3ladtueuybnu4)
- [elasticsearch-intake #28969 / part2](https://gradle-enterprise.elastic.co/s/okcj3327ugxsq)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:migrate:internalClusterTest" --tests "org.elasticsearch.xpack.migrate.action.ReindexDatastreamIndexTransportActionIT.testTsdbStartEndSet" -Dtests.seed=898E4FBF1A1263EB -Dtests.locale=kgp-BR -Dtests.timezone=Etc/GMT-6 -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.migrate.action.ReindexDatastreamIndexTransportActionIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testTsdbStartEndSet'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was completed exceptionally
```

**Issue Reasons:**
- [main] 3 failures in test testTsdbStartEndSet (1.1% fail rate in 272 executions)
- [main] 2 failures in step part2 (4.5% fail rate in 44 executions)
- [main] 2 failures in pipeline elasticsearch-intake (4.5% fail rate in 44 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -600,9 +600,6 @@ tests:
 - class: org.elasticsearch.test.rest.yaml.CcsCommonYamlTestSuiteIT
   method: test {p0=search/510_range_query_out_of_bounds/Test range query for half_float field with out of bounds upper limit}
   issue: https://github.com/elastic/elasticsearch/issues/135365
-- class: org.elasticsearch.xpack.migrate.action.ReindexDatastreamIndexTransportActionIT
-  method: testTsdbStartEndSet
-  issue: https://github.com/elastic/elasticsearch/issues/135366
 - class: org.elasticsearch.xpack.esql.session.SessionUtilsTests
   method: testFromPages
   issue: https://github.com/elastic/elasticsearch/issues/135377

--- a/x-pack/plugin/migrate/src/internalClusterTest/java/org/elasticsearch/xpack/migrate/action/ReindexDatastreamIndexTransportActionIT.java
+++ b/x-pack/plugin/migrate/src/internalClusterTest/java/org/elasticsearch/xpack/migrate/action/ReindexDatastreamIndexTransportActionIT.java
@@ -504,10 +504,7 @@ public class ReindexDatastreamIndexTransportActionIT extends ESIntegTestCase {
         """;
 
     public void testTsdbStartEndSet() throws Exception {
-        var templateSettings = Settings.builder().put("index.mode", "time_series");
-        if (randomBoolean()) {
-            templateSettings.put("index.routing_path", "metricset");
-        }
+        var templateSettings = Settings.builder().put("index.mode", "time_series").put("index.routing_path", "metricset");
         var mapping = new CompressedXContent(TSDB_MAPPING);
 
         // create template

